### PR TITLE
optimization

### PR DESCRIPTION
**Pull request by derpierre65**
-> !nexttttt, !blockabc,  doesnt work anymore
-> beat saver api improved (error handling)
-> added !unblock (#9)
-> save queue on game exit (#15)

i dont know if you want accept this pull

### CODE REVIEW

**Comment by Soliel** [comment]
@derpierre65 I'm sorry for taking so long! 

I'm gonna test these changes tomorrow then drag them in before I begin the next development cycle.
Thanks for your hard work!

File: BeatSaberTwitchIntegration/BeatBot.cs [hunk 1]
```diff
@@ -129,18 +130,15 @@ public void Initiate()
                                 var splitInput = inputLine.Split(' ');
                                 if (splitInput[0] == "PING")
                                 {
-                                    _logger.Info("Responded to twitch ping.");
                                     SendMessage("PONG " + splitInput[1]);
-                                    Thread.Sleep(1750);
+                                    continue;
```
Comment (Soliel) [inline BeatSaberTwitchIntegration/BeatBot.cs:236]: The Thread.Sleep(1750) is a lazy way to account for the twitch rate limiting. It should not be removed without an alternative being implemented.

File: BeatSaberTwitchIntegration/BeatBot.cs [hunk 2]
```diff
@@ -129,18 +130,15 @@ public void Initiate()
...
 
-                                splitInput = inputLine.Split(':');
-
-                                if (2 >= splitInput.Length) continue;
+                                // ignore server stuff there not privmsg or whisper (remove whisper?)
+                                if (splitInput[2] != "PRIVMSG" && splitInput[2] != "WHISPER") continue;
 
-                                var command = splitInput[2];
-                                OnCommandReceived(command, _currUser);
-                                Thread.Sleep(1750);
```
Comment (Soliel) [inline BeatSaberTwitchIntegration/BeatBot.cs:140]: The Thread.Sleep(1750) is a lazy way to account for the twitch rate limiting. It should not be removed without an alternative being implemented.

**Review by Soliel — CHANGES_REQUESTED** [review]
I love most of it! The only thing is getting rid of those sleeps is not good. We should have a better way of dealing with the rate limit, but until then the sleeps are all we have. 1750ms on message send stops us from going over the rate limit. Change that back and we're approved (seems kinda pendantic but setting the standard ya'know.)

File: BeatSaberTwitchIntegration/BeatBot.cs (hunk 1 above)
Comment (derpierre65) [reply BeatSaberTwitchIntegration/BeatBot.cs:236]: rate limit is with message queue solved

File: BeatSaberTwitchIntegration/BeatBot.cs (hunk 2 above)
Comment (derpierre65) [reply BeatSaberTwitchIntegration/BeatBot.cs:140]: rate limit is with message queue solved

**Review by Soliel — APPROVED** [review]

**Pull request merged**